### ISSUE TITLE
Should we add a lot more metadata to History about how the file was proccessed.

### ISSUE THREAD

**Issue by bena-nasa** [GEOS-ESM/MAPL#1919]
@lizziel @tclune @mathomp4 

Issue #1888 made me think.
1. Why are we adding metadata about the regridding method to process the data at all?
2. If we are going to add metadata about the Regridding method we should really add a lot more metadata, what is the point of adding one bit, if we don't add all the bits of how the data got from point A to B. Isn't how whether the values are time averaged vs instantaneous just a relevant? What if it was vertically regridded? Should we include metadata about that? Should we specify what the original grid was for that matter that the data was regridded from?

It just seems illogical to add to add partial metadata about the processing. Either add every detail about how it got from the original state or none, what's the point of adding just some?

**Comment by tclune** [comment]
I think the rationale is that proper handling of scientific data is to include all provenance information about how the data is generated so that someone else understands when they use the data. 

And of course there were at one time automated tools being developed that depended on _standardized_ provenance metadata.     So while I agree that we are betwixt and between, what we have now is a step in the proper direction.    A real solution would be to have a whole slew of regrid entries:   

- regrid.horizontal_method
- regrid.horizontal_other_param 
- ...
- regrid.vertical_method

**Comment by bena-nasa** [comment]
Seems like if we are taking this to the logical conclusion we should add attributes for the following as well:

1. something about the vertical Regridding
2. whether it was bit shaved
3. whether the values are instantaneous or time averaged or the min/max across the interval and if not instananeous what the averaging or period the min/max was evaluated in.
4. maybe some sort of encoding of the original grid
5. I'm probably missing something else we do...

Just seems like this is the logical conclusion that adding that one piece of metadata implies we really should be doing all the above

**Comment by tclune** [comment]
Yup.   Needs a bit of thought for organization.  Needs to be extensible as we think of more things.  And the hard bit:  check to see if there are any standards that we should be observing ...

**Comment by lizziel** [comment]
Why not make this sort of metadata be global since the information is constant per collection/file?

**Comment by tclune** [comment]
I'm wary of making it global, and technically the method can vary with in the collection.     Suppose we specify a grid for a collection and most of the requested variables are then regridded with "bilinear", but one of the requested variables is already on that grid.   The fix @mathomp4 implemented yesterday would then have "identity" for that last variable.

More importantly, there is no particular reason why different methods could not be used for different variables.  The more fundamental requirement is that all variables in a collection are on the same horizontal grid.

**Comment by mathomp4** [comment]
Yes, as @tclune says, I think the thought was that eventually (History2G?) we could maybe have per-variable regridding methods. So then you could combine things like conservative and bilinear variables in the same collection. 

Of course, at the moment it is essentially a global metadata since, per collection, everything is regridded the same! We just put it on each variable.
